synapse_spamcheck_badlist/bad_list_filter.py

```diff
@@ -52,23 +53,18 @@ def __init__(self, config, api):
         self._base_url = config["base_url"]
         logger.info("Using base url %s" % self._base_url)
 
+        # How often we should check for updates in the database.
+        self._pull_from_db_every_sec = int(config["pull_from_db_every_sec"])
+        logger.info("Rechecking database every %s seconds", self._pull_from_db_every_sec)
+
         # Regexp for extracting info from mxc links.
         self._mxc_re = re.compile("mxc://(?P<server_name>.*)/(?P<media_id>.*)")
 
-        # Linkifier, used to extract URLs from text.
-        self._linkifier = (
-            LinkifyIt()
-            .tlds(TLDS)
-            .tlds("onion", True)      # Add the `onion` tld
-            .add("git:", "http:")     # Add the `git:` scheme with the same rules as `http:`
-            .set({
-                "fuzzy_ip": True,     # Attempt to recognize e.g. 192.168.0.1
-                "fuzzy_link": True    # Attempt to recognize links without protocol
-            })
-        )
-        self._scheme_re = re.compile("https?:/*|git:/*|ftp:/*")
-        self._linkify_test_performance = Histogram('linkify_test_performance', 'Performance of calls to linkifier.test, in seconds')
-        self._link_test_performance = Histogram('link_test_performance', 'Total performance cost of checking links in a message, in seconds')
+        # A ahocorasick.Automaton used to recognize bad links.
+        self._link_automaton = None
+
+        self._link_check_performance = Histogram('synapse_spamcheck_badlist_link_check_performance', 'Performance of link checking, in seconds. This operation is in the critical path between a message being sent and that message being delivered to other members.')
+        self._md5_check_performance = Histogram('synapse_spamcheck_badlist_md5_check_performance', 'Performance of md5 checking, in seconds. This operation is in the critical path between a message being sent and that message being delivered to other members.')
```

Comment:
FWIW generally you declare the metrics as globals, as you don't want multiple instances.